### PR TITLE
Added preliminary json  output support for use with other tools

### PR DESCRIPTION
**Pull request by silvergasp**
I have added a possible implementation for a json output with description for the labels or each bounding box. Please try with person.jpg, for best demonstration of the output. I am not the python wizard so I see what I have added as a bit of a hack. But it seems to work. Please feel free to modify or give feedback for improvement. I will be using this addition regardless for a robotics project. Thankyou

### CODE REVIEW

File: net/yolov2/test.py [hunk 1]
```diff
@@ -78,14 +79,27 @@ def postprocess(self, net_out, im, save = True):
...
-			cv2.rectangle(imgcv, 
-				(left, top), (right, bot), 
+			cv2.rectangle(imgcv,
+				(left, top), (right, bot),
 				colors[max_indx], thick)
 			mess = '{}'.format(label)
-			cv2.putText(imgcv, mess, (left, top - 12), 
+			cv2.putText(imgcv, mess, (left, top - 12),
 				0, 1e-3 * h, colors[max_indx],thick//3)
+			if self.FLAGS.json:
```
Comment (thtrieu) [inline net/yolov2/test.py]: Please bring this `if` block to before line 82 and put a `continue` at the end of the block

File: net/yolov2/test.py [hunk 2]
```diff
@@ -78,14 +79,27 @@ def postprocess(self, net_out, im, save = True):
...
+						(mess, left, top, right, bot)
+				textBuff += line
 
+	# Removing trailing comma+newline adding json list terminator.
+	textBuff = textBuff[:-2] + "]"
 	if not save: return imgcv
-	outfolder = os.path.join(self.FLAGS.test, 'out') 
+	outfolder = os.path.join(self.FLAGS.test, 'out')
 	img_name = os.path.join(outfolder, im.split('/')[-1])
+	if self.FLAGS.json:
```
Comment (thtrieu) [inline net/yolov2/test.py]: Please bring this `if` block to before line `97` and add a `return` to the end of it.

**Comment by thtrieu** [comment]
Please see change request, and also does the same thing for `net/yolo/test.py` post-processing.
It would be great to add a part for JSON saving inside `README` too!

Thanks in advance :)

File: net/yolov2/test.py [hunk 3]
```diff
@@ -78,14 +79,27 @@ def postprocess(self, net_out, im, save = True):
...
+		textFile = os.path.splitext(img_name)[0] + ".json"
+		f = open(textFile, 'w')
+		f.write(textBuff)
+		f.close()
```
Comment (thtrieu) [inline net/yolov2/test.py]: a `with open(textFile, 'w') as f:` block is preferred.

**Comment by thtrieu** [comment]
Please do it with `net/yolo/test.py` as well.

**Comment by silvergasp** [comment]
Ok so I have added a section to the readme and applied changes to both `net/yolov2/test.py` and `net/yolov1/test.py` (have not tested `net/yolov1/test.py` as I don't have enough internet atm to download more training data/weights). I have also changed the file descriptor to use the with block (Didn't know that was possible :+1: thankyou!).

File: net/yolo/test.py [hunk 4]
```diff
@@ -106,16 +107,32 @@ def postprocess(self, net_out, im, save = True):
...
-	outfolder = os.path.join(FLAGS.test, 'out') 
+	# Removing trailing comma+newline adding json list terminator.
+	textBuff = textBuff[:-2] + "]"
+	outfolder = os.path.join(self.FLAGS.test, 'out')
 	img_name = os.path.join(outfolder, im.split('/')[-1])
+	if self.FLAGS.json:
+		textFile = os.path.splitext(img_name)[0] + ".json"
+		with open(textFile, 'w') as f:
+			f.write(textBuff)
+		return	if not save: return imgcv
```
Comment (thtrieu) [inline net/yolo/test.py]: oops, what's up with this line?

**Comment by thtrieu** [comment]
@silvergasp  please fix the error I commented in `yolo/test`.

**Pull request merged**